### PR TITLE
Hard-code SYCL and MPI dependency paths on installation

### PR DESCRIPTION
**Pull request by fknorr**
When the Celerity installation directory is known to CMake, it will find dependencies such as hipSYCL without the user additionally having to specify the hipSYCL installation directory.

Paths can still be explicitly overridden, but default to the installation directories discovered at configuration time. Note that this does not apply to DPC++ which requires `CMAKE_CXX_COMPILER` to be set and therefore cannot be applied from the config file.

For hipSYCL `HIPSYCL_TARGETS` is hard-coded in the same fashion, but not the deprecated `HIPSYCL_PLATFORM` / `HIPSYCL_GPU_ARCH` pair.

The CI currently cannot verify that this is correct, as it will always specify a `CMAKE_PREFIX_PATH`.

### CODE REVIEW

File: cmake/celerity-config.cmake.in [hunk 1]
```diff
@@ -11,18 +11,34 @@ include("${CELERITY_CMAKE_DIR}/AddToTarget.cmake")
...
+  endif()
 elseif(CELERITY_SYCL_IMPL STREQUAL "ComputeCpp")
+  if(NOT DEFINED ComputeCpp_DIR AND NOT "@ComputeCpp_DIR@x" STREQUAL x)
+    # ComputeCpp does not support find_dependency(PATHS)
+    set(ComputeCpp_DIR "@ComputeCpp_DIR@")
+  endif()
   find_dependency(ComputeCpp REQUIRED)
 elseif(CELERITY_SYCL_IMPL STREQUAL "DPC++")
   include("${CELERITY_CMAKE_DIR}/TestDPCXX.cmake")
   test_cxx_compiler_is_dpcpp(CXX_COMPILER_IS_DPCPP)
```
Comment (psalz) [inline cmake/celerity-config.cmake.in:39]: I'm just noticing some *slight* inconsistencies when it comes to naming, as on these three lines we use `DPC++`, `DPCXX` and `DPCPP` 😆

**Review by psalz — APPROVED** [review]
Thanks! Our developer experience is going through the roof!

**Comment by psalz** [comment]
This is not directly related to this PR, but have you managed to build Celerity against a different `MPI_HOME`? While for me it says
```
-- Found MPI_CXX: /lib-installs-local/openmpi-4.1.0/lib/libmpi.so (found suitable version "3.1", minimum required is "2.0")
```
it then goes on to say
```
CMake Warning at examples/CMakeLists.txt:19 (add_executable):
  Cannot generate a safe runtime search path for target distr_io because
  files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libmpi.so.40] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /lib-installs-local/openmpi-4.1.0/lib

  Some of these libraries may not be found correctly.
Call Stack (most recent call first):
  examples/CMakeLists.txt:56 (add_example)
```

...not sure what's happening there 🤔

File: cmake/celerity-config.cmake.in (hunk 1 above)
Comment (fknorr) [reply cmake/celerity-config.cmake.in:39]: That's why people shouldn't use special chars in product names... Well at least my `DPCXX` choice was deliberate with using `CXX` as Makefile-speak for "The DPC++ C++ compiler".

**Comment by fknorr** [comment]
> This is not directly related to this PR, but have you managed to build Celerity against a different `MPI_HOME`?

Not really. Does this happen with and without the PR?

**Comment by psalz** [comment]
> > This is not directly related to this PR, but have you managed to build Celerity against a different `MPI_HOME`?
> 
> Not really. Does this happen with and without the PR?

Yes, it also happens on current master.

**Comment by fknorr** [comment]
I've simplified some of the if-statements (`"@@x" STREQUAL x` => `"@@" STREQUAL ""`), I thought I was working around an CMake "feature" that doesn't really exist anymore.

**Review by PeterTh — APPROVED** [review]

**Pull request merged**